common/lib/xmodule/xmodule/modulestore/xml_importer.py

```diff
@@ -753,15 +753,26 @@ def _convert_ref_fields_to_new_namespace(reference):  # pylint: disable=invalid-
         # if library exists, update source_library_version and children
         # according to this existing library and library content block.
         if store.get_library(block.source_library_key):
-            LibraryToolsService(store).update_children(
-                block,
-                user_id,
-                version=block.source_library_version
-            )
+            update_block_children_on_branch(store, block, user_id, store.get_branch_setting())
 
     return block
 
 
+def update_block_children_on_branch(store, lc_block, user_id, branch):
+    """
+    Updates library content block's children on draft and published branches.
+    """
+    with store.branch_setting(branch_setting=branch):
```

Comment:
Hi, can you write up a test case that highlights the issue? `get_branch_setting` should definitely be respecting the context manager.